Authentication

IbraHiMm0 opened the pull request:
- [ ]  test

Commit:
add login screen,
add register screen,
add OTP screen,
add forgotPassword screen,
add resetPassword screen,

Commit:
add login screen,
add register screen,
add OTP screen,
add forgotPassword screen,
add resetPassword screen,

Commit:
test

Commit:
Merge remote-tracking branch 'origin/features/authentication' into features/authentication

mosayed01 commented on lib/l10n/intl_en.arb:
Please make it back @IbraHiMm0  :)

mosayed01 commented on lib/core/cubit/AuthCubit.dart:
```diff
@@ -0,0 +1,51 @@
+    selectedUniversity =newValue;
+      emit(ChangeOptionState());
+    }
+  // bool option=false;
+  // bool option1=false;
+
+
+
+  // void forgotPassOption(){
+  //   option=!option;
+  //   option1=!option;
+  //   emit(ChangeOptionForgotPassState());
+  // }
+  //
+  // void forgotPass2Option(){
+  //   option1=!option1;
+  //   option=!option1;
+  //   emit(ChangeOptionForgotPassState());
+  // }
```
please remove dead code

mosayed01 commented on lib/core/cubit/AuthCubit.dart:
```diff
@@ -0,0 +1,51 @@
+  //   emit(ChangeOptionForgotPassState());
+  // }
+
+  IconData suffix =Icons.visibility_outlined;
```
take a look if we can delegate it ti UI

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+    return Column(
+      children: [
+        Padding(
+          padding: EdgeInsets.symmetric(horizontal: width),
```
```suggestion
          padding: EdgeInsets.symmetric(horizontal: padding),
```

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+                borderRadius: BorderRadius.circular(32.r),
+                border: Border.all(color: borderColor)),
+            child: Padding(
+              padding: const EdgeInsets.only(bottom: 2.0),
```
```suggestion
              padding: EdgeInsets.only(bottom: 2.h),
```

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+              padding: const EdgeInsets.only(bottom: 2.0),
+              child: TextFormField(
+                inputFormatters: [
+                  LengthLimitingTextInputFormatter(numForm),
```
white space

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+  final Icon? icon;
+  final double width;
+  final IconButton? suffix;
+  final bool? obscureText;
```
```suggestion
  final bool obscureText;
```

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+                ],
+                keyboardType: keyboardType,
+                controller: controller,
+                obscureText: obscureText!,
```
```suggestion
                obscureText: obscureText,
```

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+                obscureText: obscureText!,
+                validator: valid,
+                cursorColor: EVColors.primaryLight,
+                style:  TextStyle(
```
use figma text styles with copyWith to change color

mosayed01 commented on lib/core/widget/primary_textfield.dart:
```diff
@@ -0,0 +1,95 @@
+                  focusedErrorBorder: InputBorder.none,
+                  prefixIcon: icon,
+                  hintText: hintText,
+                  hintStyle:  TextStyle(
```
use figma text styles with copyWith to change color

mosayed01 commented on lib/presentation/forgotPassword/forgotPassword_screen/forgotPassword_screen.dart:
```diff
@@ -0,0 +1,149 @@
+
+              SvgPicture.asset(
+                Assets.book,
+                width: 64.w,
+                height: 64.h,
```
```suggestion
                width: 64.w,
                height: 64.w,
```

mosayed01 commented on lib/presentation/forgotPassword/forgotPassword_screen/forgotPassword_screen.dart:
```diff
@@ -0,0 +1,149 @@
+                    borderRadius: BorderRadius.circular(35.r),
+                  ),
+                  child: Padding(
+                    padding: const EdgeInsets.all(12.0),
```
```suggestion
                    padding: EdgeInsets.all(12.w),
```

mosayed01 commented on lib/presentation/login/login_screen/login_screen.dart:
```diff
@@ -0,0 +1,221 @@
+                      },
+                      keyboardType: TextInputType.emailAddress,
+                    ),
+                    verticalSpacer(20),
```
review figma spaces and dimens

mosayed01 reviewed (changes requested):
GG

Commit:
Merge remote-tracking branch 'origin/features/authentication' into features/authentication

# Conflicts:
#	lib/core/route/app_router.dart
#	lib/core/theme/assets.dart
#	lib/l10n/intl_en.arb
#	pubspec.yaml

Commit:
add login screen,
add register screen,
add OTP screen,
add forgotPassword screen,
add resetPassword screen,
add strategies screen,
add courseDetails screen,
add notifications ,
add settings screen,

Commit:
Merge remote-tracking branch 'origin/develop' into features/authentication

# Conflicts:
#	assets/images/alarm.png
#	lib/core/theme/assets.dart

Commit:
add navigation
